src/identity/anonymous/anonymous-identity.ts

```diff
@@ -0,0 +1,26 @@
+import { Identity } from "../types"
+import { ANONYMOUS, CoseKey, EMPTY } from "../../message/cose"
+
+export class AnonymousIdentity extends Identity {
+  publicKey = ANONYMOUS
+  async sign() {
+    return EMPTY
+  }
+  async verify() {
+    return false
+  }
+
+  getCoseKey(): CoseKey {
+    const c = new Map()
```

Comment:
?

Reply:
`this.getCommon` was removed from `CoseKey`, moved into `getCoseKey` for each Identity
code snip below is from current `main` branch and for `ANONYMOUS`, we still create the common structure

```
// this.getProtectedHeader (ANONYMOUS) -> new CoseKey(ANONYMOUS) -> getCommon(ANONYMOUS)

    const protectedHeader = this.getProtectedHeader(
      keys ? keys.publicKey : ANONYMOUS
    );
    
    private static getProtectedHeader(publicKey: Key): CborMap {
      const coseKey = new CoseKey(publicKey);
      const protectedHeader = new Map();
      protectedHeader.set(1, -8); // alg: "Ed25519"
      protectedHeader.set(4, coseKey.keyId); // kid: kid
      protectedHeader.set("keyset", coseKey.toCborData());
      return protectedHeader;
    }


export class CoseKey {
  key: CborMap;
  keyId: CborData;
  private common: CborMap;

  constructor(publicKey: Key) {
   // publicKey would be ANONYMOUS here
    this.common = this.getCommon(publicKey);
    this.keyId = this.getKeyId();
    this.key = this.getKey();
  }

  private getCommon(publicKey: Key) {
    const common = new Map();
    common.set(1, 1); // kty: OKP
    common.set(3, -8); // alg: EdDSA
    common.set(-1, 6); // crv: Ed25519
    common.set(4, [2]); // key_ops: [verify]
    common.set(-2, publicKey); // x: publicKey
    return common;
  }
```

Reply:
> Weird thing about the anonymous identity having a public key...
public key has been removed from anon identity.